test/test_enforced_encryption.cpp

```diff
@@ -348,7 +348,7 @@ class TestEnforcedEncryption
         const TResult &expect = test.expected_result;
 
         // Start testing
-        volatile bool caller_done = false;
```

Comment:
This variable is marked `volatile` for the compiler not to optimize it out in the loop inside the thread:
```c++
do
{
    std::this_thread::sleep_for(std::chrono::milliseconds(50));
} while (!caller_done);
```

It does not necessarily need to be atomic, as it is only changed in one thread. Missing that change is not too bad, as it would just cause another 50ms of waiting.

Reply:
If it's `atomic`, it still won't be optimized out by compiler, and it's more correct in semantic.